docs/advanced.rst

```diff
@@ -457,6 +457,119 @@ higher round-trip-time (RTT) lead to longer installation times.
 This can be compensated somewhat by using a HTTP/2 server, as this supports
 multiplexing and better connection reuse.
 
+.. _sec-encryption:
+
+Bundle Encryption
+-----------------
+
+RAUC supports encrypting the bundle to one or more recipients (public keys).
+
+The implementation of the crypt bundle format is based on the verity bundle
+format (which uses Linux's dm-verity module).
+It works by symmetrically encrypting the bundle payload and using
+Linux's dm-crypt module to decrypt this on-demand.
+The symmetric encryption key is contained in the manifest, which
+itself is (asymmetrically) encrypted to a set of recipients.
+Similar to the verity format, the crypt format can also be used
+with HTTP streaming.
+
+To use encryption, some prerequisites need to be fulfilled:
+
+- create bundle using the crypt format
+- enable dm-crypt support in the target's kernel
+
+Creating an encrypted bundle has two main steps:
+
+- encrypting the payload with ``rauc bundle`` using a manifest configured for the crypt format
+- encrypting the manifest with the payload encryption key for specific recipients with ``rauc encrypt``
+
+We've separated these steps to support more flexibility regarding decryption keys.
+Some possible workflows are described in :ref:`sec-encryption-workflows`.
+
+The first step can be performed by a build system, very similar to how un-encrypted bundles are created.
+RAUC generates a random key for symmetric AES-256 encryption of the bundle payload (the SquashFS).
+The encrypted payload is then protected against modification with dm-verity (see the verity format for details).
+The AES key is stored (*as plain text*) in the signed manifest.
+
+The second step needs to be performed before publishing the bundle.
+You need to provide (one or more) recipient certificates,
+which are used to encrypt the signed manifest.
+The already encrypted payload is reused unmodified.
+Any of the corresponding private keys can then be used by RAUC to first decrypt the
+manifest, which then contains the key needed to decrypt the (SquashFS) payload.
+
+.. code-block::
+
+   $ rauc encrypt --to=recipient-certs.pem unencrypted-crypt-bundle.raucb encrypted-crypt-bundle.raucb
+
+.. note::
+   To encrypt for a larger number of recipients, the recipient certificates can be
+   concatenated and provided as a single file in the ``--to`` argument.
+
```

Comment:
We should mention here that the recipient certificates don't need to be part of the signing PKI.

Reply:
I've dropped a note. Also realized that documentation might still be quite incomplete when one actually wants to set up encryption. I'll add some further hints about usage and target configuration